types/nodepub/index.d.ts

```diff
@@ -0,0 +1,75 @@
+// Type definitions for nodepub 3.0
+// Project: https://github.com/kcartlidge/nodepub
+// Definitions by: Dylan Armstrong <https://github.com/dylanarmstrong>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface Links {
+  itemType: 'front' | 'contents' | 'main';
+  link: string;
+  title: string;
+}
+
+export type GenerateContentsCallback = (links?: Links) => string | void;
```

Comment:
```suggestion
export type GenerateContentsCallback = (links: Link[]) => string | void;
```
it's required and it takes list of objects:
https://github.com/kcartlidge/nodepub#you-can-also-choose-to-make-your-own-custom-contents-page

Reply:
Ah, I actually thought that too but then I saw the test:
https://github.com/kcartlidge/nodepub/blob/master/test/index-spec.js#L125

Reply:
https://github.com/kcartlidge/nodepub/blob/master/src/constituents/markup.js#L18

There's also a check that it's defined. So, I think it could be called with an undefined argument and return undefined in the types.

Reply:
hey, yes, because the list could be empty, so the outcome would evalute to false in expression check.
But it's required here:
https://github.com/kcartlidge/nodepub/blob/6b9c599d0b224947362222c23598c62c0d848a1e/src/constituents/markup.js#L40
and files are always passed:
https://github.com/kcartlidge/nodepub/blob/6b9c599d0b224947362222c23598c62c0d848a1e/src/index.js#L37

Reply:
Ok, thanks that makes sense. I think this is ready for re-review then.